addons/stock/models/stock_location.py

```diff
@@ -232,16 +238,14 @@ def write(self, values):
             modified_locations = self.filtered(
                 lambda l: any(l[f] != values[f] if f in values else False
                               for f in {'usage', 'scrap_location'}))
-            reserved_quantities = self.env['stock.move.line'].search_count([
+            reserved_quantities = self.env['stock.quant'].search_count([
                 ('location_id', 'in', modified_locations.ids),
-                ('state', 'not in', ['done', 'cancel']),
-                ('quantity_product_uom', '>', 0),
+                ('location_id.usage', '=', 'internal'),
```

Comment:
There is already a condition on `self.ids`. Why adding a condition on usage ?

Reply:
mybad, yaa no need of that condition.